core/src/main/java/org/springframework/security/access/expression/method/MethodSecurityEvaluationContext.java

```diff
@@ -15,105 +15,38 @@
  */
 package org.springframework.security.access.expression.method;
 
-import java.lang.reflect.Method;
-
 import org.aopalliance.intercept.MethodInvocation;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.springframework.aop.framework.AopProxyUtils;
 import org.springframework.aop.support.AopUtils;
+import org.springframework.context.expression.MethodBasedEvaluationContext;
 import org.springframework.core.ParameterNameDiscoverer;
-import org.springframework.expression.spel.support.StandardEvaluationContext;
 import org.springframework.security.core.Authentication;
-import org.springframework.security.core.parameters.DefaultSecurityParameterNameDiscoverer;
+
+import java.lang.reflect.Method;
 
 /**
  * Internal security-specific EvaluationContext implementation which lazily adds the
  * method parameter values as variables (with the corresponding parameter names) if and
  * when they are required.
  *
  * @author Luke Taylor
+ * @author Daniel Bustamante
  * @since 3.0
  */
-class MethodSecurityEvaluationContext extends StandardEvaluationContext {
+class MethodSecurityEvaluationContext extends MethodBasedEvaluationContext {
 	private static final Log logger = LogFactory
 			.getLog(MethodSecurityEvaluationContext.class);
 
-	private ParameterNameDiscoverer parameterNameDiscoverer;
-	private final MethodInvocation mi;
-	private boolean argumentsAdded;
-
-	/**
-	 * Intended for testing. Don't use in practice as it creates a new parameter resolver
-	 * for each instance. Use the constructor which takes the resolver, as an argument
-	 * thus allowing for caching.
-	 */
-	public MethodSecurityEvaluationContext(Authentication user, MethodInvocation mi) {
-		this(user, mi, new DefaultSecurityParameterNameDiscoverer());
-	}
```

Comment:
Is it necessary to remove this constructor? Although it is only intended for testing, removing it would not be backward compatible

Reply:
Hi @eleftherias I'll proceed to check the comment and rebase, thanks for your comment.

Reply:
@dbuos Have you had the chance to review the requested changes?